### ISSUE TITLE
AsJobs timeout and tries

### ISSUE THREAD

**Issue by exodusanto** [lorisleiva/laravel-actions#81]
Hi,

i'm upgrading from v1 and I can't find in the docs where to place the `timeout` and `tries` property / option for a job.

```php
    /**
     * The number of seconds the job can run before timing out.
     *
     * @var int
     */
    public $timeout = 6000; // default: 240

    /**
     * The number of times the job may be attempted.
     *
     * @var int
     */
    public $tries = 25; // default: 5
```
Digging in `configureJob` and `JobDecorator` i found the `delay` or `queueName` but no the `timeout` or `tries`.

**Comment by lorisleiva** [comment]
Hi there 👋 

Thanks for reporting this. It must be an overlook on my side. I'm working on it. 🙂

**Comment by exodusanto** [comment]
I found also another 2 issues 😅

### AsCommand

Using scheduler

```php
$scheduler->command(MyAction::class);
```

Scheduler fire this exception:

`Call to undefined method App\Actions\MyAction::getName()`

### AsJob

Adding `Batchable` trait queue throw this error:

`Error: Call to undefined method Lorisleiva\Actions\Decorators\JobDecorator::withBatchId()`

**Comment by lorisleiva** [comment]
Awesome! I'll tackles these as well today.

Feel free to keep adding bugs on this issues as you find them. It's always so scary to launch something and then have zero feedback haha. Thanks!

**Comment by exodusanto** [comment]
I will 😁, this new design pattern of v2 it's pretty dope 👍🏻

**Comment by lorisleiva** [comment]
First bug fixed and released as [`v2.0.1`](https://github.com/lorisleiva/laravel-actions/releases/tag/v2.0.1).

I've also [updated the documentation](https://laravelactions.com/2.x/dispatch-jobs.html#configuring-jobs).

I've added some tests but please let me know if that works for you to double-check. 🙂 

Onto the next one. 💪

**Comment by exodusanto** [comment]
Sure!

If I want to use with `configureJob` is this way correct? Or u suggest to use the `jobTimeout` and `jobTries` properties?

```php
    public function configureJob(JobDecorator $job): void
    {
        $job->timeout = 6000;
        $job->tries = 25;
    }

```

**Comment by lorisleiva** [comment]
The issue regarding the scheduled command has been fixed in [`v2.0.2`](https://github.com/lorisleiva/laravel-actions/releases/tag/v2.0.2).

Yes, that's correct but we can do better than that. I've added the following setters on the `JobDecorator` (in `v2.0.2` as well). 🙂 

```php
public function configureJob(JobDecorator $job): void
{
    $job->setTries(10)
        ->setMaxExceptions(3)
        ->setTimeout(60 * 30);
}
```

**Comment by lorisleiva** [comment]
The last one about dispatchable jobs has been released in [`v2.0.3`](https://github.com/lorisleiva/laravel-actions/releases/tag/v2.0.3). 🚀 

I'm closing this now that everything has been handled but please don't hesitate to open new issues.

Thanks again for your help. 🙂

**Comment by exodusanto** [comment]
Great!

Another thing 😌, if I need to use the `JobDecorator` inside the `asJob`?

```php

    public function asJob($user): void
    {
        if ($this->batch() && $this->batch()->cancelled()) {
            return;
        }

        $this->handle(User::find($user));
    }
```
